src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,130 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double ADD_CAPACITY = 1.5;
+    private static final int POSITION_ZERO = 0;
+    private T [] defaultArray;
+    private int listSize;
+
+    public ArrayList() {
+        defaultArray = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private void isIndexInBounds(int index) {
+        if (!(index >= 0 && index < listSize)) {
+            throw new ArrayListIndexOutOfBoundsException("Index: "
+                    + index
+                    + " is out of bounds. "
+                    + "Size: "
+                    + listSize);
+        }
```

Comment:
The method 'isIndexInBounds' could be renamed to be more descriptive and indicate that it checks if the index is within the valid range for existing elements. Consider renaming it to something like 'checkIndexForExistingElements'.